### DISCUSSION TITLE
Redundant Fields in Forms?

### DISCUSSION THREAD

**Discussion by pand0r** [sonata-project/SonataAdminBundle#7935] [Q&A]
Hello!
I’m really new into sonata and have a question about the forms.
Is it possible to add static elements to a form?
I have to build a form form with multiple tabs. On the first tab I have an editable field for a file number. This file number has to be shown in all other tabs too, as non editable plaintext.

Question 1:
Is there any way to show that filed as plaintext other than set the form-fields as disabled + add a custom css-class to remove the look of an input-field?

Question 2:
To show that field multiple times, my workaround is that i create duplicates of the getter/setter for that field in my entity and use them for the show it in the form.

**Comment by eerison** [comment]
Hi @pand0r have you checked the Sonata docs? 

here you have the option to create groups: https://docs.sonata-project.org/projects/SonataAdminBundle/en/4.x/reference/action_show/?highlight=group#group-options

Maybe it can help you!

and for fields not mapped, you have a option `'mapped' => false,`

e.g: 

```php
            ->add('your field', null, [
                    'mapped' => false,
                ])
```

**Comment by VincentLanglet** [comment]
> Is there any way to show that filed as plaintext other than set the form-fields as disabled + add a custom css-class to remove the look of an input-field?
> To show that field multiple times, my workaround is that i create duplicates of the getter/setter for that field in my entity and use them for the show it in the form.

I recommend you to take a look at the `TemplateType`, it should do exactly what you want
https://docs.sonata-project.org/projects/SonataAdminBundle/en/4.x/reference/action_create_edit/#displaying-custom-data-template
